frontend/workflows/experimentation/src/list-experiments.tsx

```diff
@@ -1,49 +1,73 @@
 import React, { useState } from "react";
 import { useNavigate } from "react-router-dom";
 import type { clutch as IClutch } from "@clutch-sh/api";
-import type { BaseWorkflowProps } from "@clutch-sh/core";
 import { ButtonGroup, client, Error, Row, Table } from "@clutch-sh/core";
 import { Container } from "@material-ui/core";
 import styled from "styled-components";
 
 interface ExperimentationSpecificationDataProps {
   experiment: IClutch.chaos.experimentation.v1.Experiment;
+  columns: string[];
+  experimentTypes: any;
 }
 
 const ExperimentSpecificationData: React.FC<ExperimentationSpecificationDataProps> = ({
   experiment,
+  columns,
+  experimentTypes,
 }) => {
-  const specification = experiment.testSpecification;
-  const data = specification?.abort ? specification.abort : specification.latency;
-
-  const defaultData = [
-    data.clusterPair.downstreamCluster,
-    data.clusterPair.upstreamCluster,
-    data.percent,
-  ];
-  if (specification?.abort) {
-    defaultData.push((data as IClutch.chaos.experimentation.v1.AbortFault).httpStatus);
+  const types = experimentTypes || {};
+  const registeredExperimentType = types[experiment.testConfig["@type"]];
+  const isExperimentTypeRegistered = typeof registeredExperimentType !== "undefined";
+  if (!isExperimentTypeRegistered) {
+    const data = columns.map(() => {
+      return experiment.testConfig["@type"];
+    });
+    return <Row data={data} />;
   }
 
-  return <Row data={defaultData} />;
+  const mapper = registeredExperimentType.mapping;
+  const mapperExists = typeof mapper !== "undefined";
```

Comment:
Ditto using `registeredExperimentType.hasOwnProperty(registeredExperimentType.mapping)`, but check with @dschaller  before taking my word on it.

Reply:
👍